modules/core/src/main/java/org/apache/ignite/startup/cmdline/CommandLineStartup.java

```diff
@@ -66,6 +86,48 @@
     /** Quite log flag. */
     private static final boolean QUITE;
 
+    /** Command to print Ignite system properties info. */
+    static final String PRINT_PROPS_COMMAND = "-systemProps";
+
+    /** Classes with Ignite system properties. */
+    static final List<Class<?>> PROPS_CLS = new ArrayList<>(Arrays.asList(
+        IgniteSystemProperties.class,
```

Comment:
It seems some of the classes missed in this list:
- H2Tree
- ZookeeperDiscoveryImpl

Should we add checks here at compile time to ensure that this list is fully enough? Some of the developers may forgot adding new classes here. Also, we should to mention it in javadoc of the `@SystemProperty` annotation.

Reply:
What checks do you have in mind?

Reply:
I don't have a simple solution for this case, probably, some autogenerated resources might help here. As another option, we can write a custom checkstyle rule but it less friendly for the developers.

Reply:
Let's postpone this task to another ticket?

I agree with you that this should be checked automatically, but, without simple answers, we have to think and come to the dev-list with some solution

Reply:
OK, let's file a ticket right after this one will be merged.

Reply:
IGNITE-13421 created